File: Python/06.py
'''

Project Euler Solutions - Problem 06

By Chris Mander (http://github.com/manderc3)

'''

import time

def differenceInSquareSums(n):
	sum_of_squares, sum_of_nums = 0, 0
	
	for i in range(1, n + 1):
		sum_of_squares += i ** 2
		sum_of_nums += i

	return sum_of_nums ** 2 - sum_of_squares

print(differenceInSquareSums(100))

File: c/06.c
#include <stdio.h>
#include <stdlib.h>

int toThePower(int base, int exponent) {
	int i, result = 1;

	for (i = 0; i < exponent; i++) {
		result *= base;
	}

	return result;
}

unsigned int differenceInSquareSums(int n) {
	int sum_of_squares = 0, sum_of_nums = 0, i;

	for (i = 1; i < n + 1; i++) {
		sum_of_squares += toThePower(i, 2);
		sum_of_nums += i;
	} 

	return toThePower(sum_of_nums, 2) - sum_of_squares; 
}

int main() {

	printf("%d\n", differenceInSquareSums(100));
	return 0;
}